src/main/java/io/anserini/util/PrebuiltIndexHandler.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Anserini: A Lucene toolkit for reproducible information retrieval research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.util;
+
+import me.tongfei.progressbar.ProgressBar;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.CountingInputStream;
+
+import io.anserini.index.IndexInfo;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class PrebuiltIndexHandler {
+  private String indexName;
+  private String saveRootPath;
+  private IndexInfo info = null;
+  private Path indexFolderPath = null;
+  private boolean initialized = false;
+  private Path savePath;
+
+  // get cache path, we use pyserini path to avoid double download
+  String CACHE = Paths.get(System.getProperty("user.home"), ".cache", "pyserini", "indexes").toString();
```

Comment:
How's this: look first in `~/.cache/pyserini/indexes` - if there, use it. Otherwise, download to `~/.cache/anserini/indexes`?